### DISCUSSION TITLE
Pihole einfach umgangen mit ändern des DNS Servers in Ubuntu?!?!

### DISCUSSION THREAD

**Discussion by bejuicey** [RPiList/specials#1131] [Q&A] [answered]
Hallo zusammen,

ich habe das Pihole Installiert und es funktioniert!
Ich bin davon ausgegangen das ich den DNS Server nicht umgehen kann.
Jetzt mache ich einen LPIC 1 Kurs und bei den Netzwerkeinstallungen hat sich herausgestellt,
das wenn ich den DNS Server in Ubuntu auf 8.8.8.8 (in der VM, DHCP ist eingeschaltet im Router) stelle, jetzt das Pihole einfach umgangen wird?!?!

Problem:
Ich .. P .. Problem ... die sperre ich für mich, der DNS Server und Router ist abgeschlossen im Schrank damit
Ich nicht rankomme. Beim LPIC-1 Kurz hat sich herausgestellt mit ein paar klicks habe ich den DNS Server umgangen
-> Nicht gut, dachte das geht nicht.

  
Setup:
Gigacube LTE Router -> Fritzbox 6490 Cable ( Zugang über Lan1), DNS Server über Pihole aktiviert.

Frage:
Wie kann ich die DNS umgehung sperren? Mit Squid auf einem weiteren Pihole und das Internet allgmein sperren für alle Geräte außer dem DNS Server und Proxy?

LG

**Comment by LizenzFass78851** [comment, marked as answer]
Vielleicht hilft dir dieses Video von @RPiList weiter: https://youtu.be/AIwK4URbSv8

**Reply by bejuicey** [reply]
Nein das Hat leider nicht geholfen, hier ein paar Screenshots.
Sorry im Lost, ich habe das jetzt ausgiebig getestet aber es will nicht laufen.

![Interneteinstellungen](https://github.com/RPiList/specials/assets/138672588/14140d3b-902a-4459-af51-8615ffd83abd)
![Heimnetzeinstellungen](https://github.com/RPiList/specials/assets/138672588/1e85ed62-074d-444f-8fd6-94ea3422041e)
![DNS-Server](https://github.com/RPiList/specials/assets/138672588/a09c0ac7-e53d-459e-a069-dc4f3b434ff7)

![DNSsperre53upd](https://github.com/RPiList/specials/assets/138672588/dc77d38e-6435-4732-ad78-b2c6e8ee3d8f)
![Kindersicherung](https://github.com/RPiList/specials/assets/138672588/3ff3a9ed-5cd3-4937-b183-35573b66c75e)
![Desktop mit aktueller DNS](https://github.com/RPiList/specials/assets/138672588/12f90e2f-eb4a-46e8-bdc3-1b4c72b9b9ff)
![Sollte nicht gehen](https://github.com/RPiList/specials/assets/138672588/ca81500c-dc47-4b86-ad63-6448e0da3728)
![Zugangsprofile](https://github.com/RPiList/specials/assets/138672588/27055f6d-de74-468c-821a-ade796ebf591)

**Reply by LizenzFass78851** [reply]
Probiere in ein Terminal den Befehl `dig google.com`.
Wenn in der Ausgabe des Befehls was von `;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)` (die IP 127.0.0.53 aus den Bereich der Localhost IP) steht dann Versuche den [systemd resolved zu deaktivieren](https://askubuntu.com/questions/907246/how-to-disable-systemd-resolved-in-ubuntu) (Funtioniert auch auf neueren Ubuntu bassierten Systemen und Linux Mint)

Wenn der systemd resolved deaktiviert ist, sollte mit den Befehl `dig google.com` die Antwort von den per DHCP oder den selbst auf dem System definierten DNS Server kommen. z.B. `;; SERVER: 192.168.178.15#53(192.168.178.15) (UDP)`

Andernfalls mit den Befehl `dig google.com @1.1.1.1` prüfen ob das System überhaupt eine DNS Verbindung nach außen aufbauen kann.
Wenn als Ausgabe:
```
;; communications error to 1.1.1.1#53: host unreachable
.....
;; no servers could be reached
```
kommt, dann sollte die DNS Sperrung über die FritzBox funktionieren.

Alternativ wenn das nichts hilft, die FritzBox neustarten und oder im Terminal den Befehl `sudo dhclient -r && sudo dhclient && sudo systemctl restart networking && sudo systemctl restart NetworkManager` den DHCP-Request neu abfragen.

**Reply by bejuicey** [reply]
Das hat jetzt geholfen, darf ich fragen was jetzt genau passiert ist um das zu verstehen?

**Reply by bejuicey** [reply]
Bzw. über welchen Port arbeitet systemd-resolve?

**Reply by LizenzFass78851** [reply]
Bei Ubuntu basierten Systemen und Linux Mint o. Ä. Gibt es den (vorinstallierten) [systemd resolved](https://wiki.ubuntuusers.de/systemd/systemd-resolved/) der (laut wiki.ubuntuusers.de) für folgendes sorgt:
- Bereitstellung eines lokalen DNS-Cache: Dies ist ein eigenständiger DNS-Server, der jedoch selbst keine Namenszuordnungen bereit stellt, sondern nur die von anderen DNS-Servern (Forwarder) im Netz erhaltenen DNS-Antworten speichert und damit für den Anwender das Zeitverhalten bei DNS-Abfragen verbessert.
- Auflösung von Rechnernamen zu IP-Adressen per Unicast-DNS, LLMNR und MulticastDNS
- Veröffentlichung von Rechnernamen im lokalen Netz per LLMNR und MulticastDNS 

Vermutlich scheint der systemctl resolved was in der Hinsicht zu machen was für den Fehler (aus deinen Screenshots) verantwortlich ist. Wie genau der resolved den Fehler verursacht weiß ich leider nicht. 

Bei Debian gibt es evtl. den systemctl resolved nicht da er bei Debian ggf. nicht vorinstalliert ist.
